codegenerator/cli/templates/dynamic/codegen/src/Types.res.hbs

```diff
@@ -532,48 +531,50 @@ module MakeRegister = (Event: Event) => {
     )
 }
 
+type fuelEventKind = | Mint | LogData({
+  logId: string,
+  decode: string => internalEventArgs,
+})
+
+type fuelEventConfig = {
+  name: string,
+  kind: fuelEventKind,
+  isWildcard: bool,
+  handlerRegister: HandlerTypes.Register.t<internalEventArgs>,
+  paramsRawEventSchema: S.schema<internalEventArgs>,
+}
+
+type fuelContractConfig = {
+  name: string,
+  events: array<fuelEventConfig>,
+}
```

Comment:
Moved it here to prevent cycle dependency

Reply:
Where was this moved from? It looks like new code?

Reply:
True, I've written it on Thursday, so it's already old for me 😅